src/vcf/validate_optional_policy.cpp

```diff
@@ -146,6 +152,36 @@ namespace ebi
         }
     }
 
+    void ValidateOptionalPolicy::check_body_entry_alt_gvcf_end(ParsingState & state, Record const & record) const
+    {
+        if (std::find(record.alternate_alleles.begin(), record.alternate_alleles.end(), GVCF) != record.alternate_alleles.end()
+            && record.info.find(END) == record.info.end()) {
+            throw new AlternateAllelesBodyError{state.n_lines,
+                    "INFO END should be provided, as " + GVCF + " (symbolic alternate allele) is supposed to represent a region"};
+        }
+    }
+
+    void ValidateOptionalPolicy::check_body_entry_alt_gvcf_gt_value(ParsingState & state, Record const & record) const
+    {
+        if (std::find(record.alternate_alleles.begin(), record.alternate_alleles.end(), GVCF) != record.alternate_alleles.end()
+            && record.format[0] == vcf::GT) {
+            size_t provided_ploidy = state.source->ploidy.get_ploidy(record.chromosome);
+            std::string required_gt_subfield = "0";
+            while (--provided_ploidy) {
+                required_gt_subfield += "/0";
+            }
+            for (auto & sample : record.samples) {
+                if (sample.substr(0, required_gt_subfield.size() + 1) == required_gt_subfield + ":"
+                    || sample == required_gt_subfield) {
```

Comment:
switch this `||` as it will read positions past the end of the string if `sample == required_gt_subfield` is true